api/src/search/service.ts

```diff
@@ -25,32 +26,70 @@ export class SearchService {
     });
   }
 
-  public search = async (query: string): Promise<SearchItem[]> => {
+  public search = async (query: string): Promise<SearchResults> => {
     this.logger.info({ message: `Searching for ${query}` });
-    return [];
+    return {
+      results: [],
+    };
   };
 
-  public index = async (
+  public upsert = async <T extends BaseEntity>(
     index: SearchType,
-    data: SearchItem[],
+    data: T,
   ): Promise<void> => {
     this.logger.info({
-      message: `Indexing ${data.length} items in ${index}`,
+      message: `Upserting "${data.id}" item to ${index}`,
+    });
+    await this.meilisearch.index(index).updateDocuments([data]);
+    this.logger.info({ message: `Upserted "${data.id}" item to ${index}` });
+  };
+
+  public deleteAllButWithRunId = async (
+    index: SearchType,
+    runId: string,
+  ): Promise<void> => {
+    this.logger.info({
+      message: `Deleting all ${index} but with runId ${runId}`,
+    });
+    await this.meilisearch.index(index).deleteDocuments({
+      filter: `NOT runId=${runId}`,
     });
-    await this.meilisearch.index(index).addDocuments(data);
     this.logger.info({
-      message: `Indexed ${data.length} items in ${index}`,
+      message: `Deleted all ${index} but with runId ${runId}`,
     });
   };
 
-  public ensureIndexes = async (): Promise<void> => {
-    await this.meilisearch.createIndex("project");
+  public setupSearch = async (): Promise<void> => {
+    await this.setupIndexes();
+    await this.updateFilterableAttributes();
+  };
+
+  private setupIndexes = async (): Promise<void> => {
+    await this.meilisearch.createIndex("project", {
+      primaryKey: "id",
+    });
```

Comment:
`createIndex` will throw if the index already exists, let's create our own `upsertIndex` method and call it instead:

```ts
  private upsertIndex = async (index: string): Promise<void> => {
    try {
      await this.meilisearch.getIndex(index);
      this.logger.info({ message: `${index} index already exists` });
    } catch {
      await this.meilisearch.createIndex(index, { primaryKey: "id" });
      this.logger.info({ message: `${index} index created` });
    }
  };
```

```suggestion
    await this.upsertIndex("project");
```